tools/run_tests/run_tests_matrix.py

```diff
@@ -536,6 +541,8 @@ def _runs_per_test_type(arg_str):
         extra_args.append('--bq_result_table')
         extra_args.append('%s' % args.bq_result_table)
         extra_args.append('--measure_cpu_costs')
+    if args.extra_args:
+        extra_args.append('%s' % args.extra_args)
```

Comment:
extra_args is a list and you are adding all extra args as just one item of the list. (`-r foo` will be added as just one arg, but it should be added as two separate items to the list: `['-r', 'foo']`. If this works, it's more of a concidence.